src/controllers/import.js

```diff
@@ -10,85 +10,152 @@
+      throw new ErrorWithStatusCode('Invalid request: request body data is required', STATUS_BAD_REQUEST);
     }
 
-    if (!Array.isArray(data.urls)) {
-      throw new ErrorWithStatusCode('Invalid request: urls must be provided as an array', BAD_REQUEST);
+    if (!Array.isArray(data.urls) || !data.urls.length > 0) {
+      throw new ErrorWithStatusCode('Invalid request: urls must be provided as a non-empty array', STATUS_BAD_REQUEST);
     }
 
     data.urls.forEach((url) => {
       if (!isValidUrl(url)) {
-        throw new ErrorWithStatusCode(`Invalid request: ${url} is not a valid URL`, BAD_REQUEST);
+        throw new ErrorWithStatusCode(`Invalid request: ${url} is not a valid URL`, STATUS_BAD_REQUEST);
       }
     });
 
     if (data.options && !isObject(data.options)) {
-      throw new ErrorWithStatusCode('Invalid request: options must be an object', BAD_REQUEST);
+      throw new ErrorWithStatusCode('Invalid request: options must be an object', STATUS_BAD_REQUEST);
     }
   }
 
   function validateImportApiKey(importApiKey) {
     // Parse the allowed import keys from the environment
-    const allowedImportApiKeys = env.ALLOWED_IMPORT_API_KEYS?.split(',') || [];
+    const allowedImportApiKeys = IMPORT_ALLOWED_API_KEYS?.split(',') || [];
     if (!allowedImportApiKeys.includes(importApiKey)) {
       throw new ErrorWithStatusCode('Invalid import API key', 401);
     }
   }
 
+  function createErrorResponse(error) {
+    return createResponse({}, error.status || 500, {
+      [HEADER_ERROR]: error.message,
+    });
+  }
+
   /**
-   * Create a new import job.
-   * @param requestContext
-   * @returns {Promise<*>}
+   * Create and start a new import job.
+   * @param {object} requestContext - Context of the request.
+   * @param {Array<string>} requestContext.data.urls - Array of URLs to import.
+   * @param {object} requestContext.data.options - Optional import configuration parameters.
+   * @param {string} requestContext.pathInfo.headers.x-import-api-key - API key to use for the job.
+   * @returns {Promise<Response>} 202 Accepted if successful, 4xx or 5xx otherwise.
    */
   async function createImportJob(requestContext) {
     const { data, pathInfo: { headers } } = requestContext;
-    const importApiKey = headers['x-import-api-key'];
+    const { 'x-import-api-key': importApiKey } = headers;
 
     try {
-      validateRequestData(data);
       validateImportApiKey(importApiKey);
+      validateRequestData(data);
+
+      const { urls, options } = data;
+      const job = await importSupervisor.startNewJob(urls, importApiKey, options);
 
-      // const { urls, options } = data;
-      // const jobResponse = await importSupervisor.startNewJob(urls, options, importApiKey);
-      return createResponse({}, 501);
+      return createResponse(ImportJobDto.toJSON(job), STATUS_ACCEPTED);
     } catch (error) {
       log.error(`Failed to queue import job: ${error.message}`);
```

Comment:
We don't have access to the job here yet since the error would likely occur in `importSupervisor.startNewJob(..)`. I've added a bit more logging to this method to aid in debugging.